Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCreateTransmission.py

```diff
@@ -0,0 +1,179 @@
+        """If a flood workspace has been provided then apply a flood correction"""
+        flood_ws = self.getProperty(Prop.FLOOD_WS).value
+        if flood_ws is None:
+            return workspace
+
+        self.log().information("Performing flood correction")
+        args = {"FloodWorkspace": flood_ws}
+        corrected_ws = self._run_algorithm(workspace, self._FLOOD_ALG, "InputWorkspace", args)
+
+        # The flood correction performs a divide, which can result in the workspace being set as a distribution
+        # if the flood and input workspaces have the same Y units. We need to set the resulting workspace back to
+        # a non-distribution as it may be dimensionless after the divide, but it shouldn't be a distribution.
+        if isinstance(corrected_ws, WorkspaceGroup):
+            for child_ws in corrected_ws:
+                child_ws.setDistribution(False)
+        else:
+            corrected_ws.setDistribution(False)
+
+        return corrected_ws
+
+    def _apply_background_subtraction(self, workspace):
+        """If background processing instructions have been provided then perform a background subtraction"""
+        if self.getProperty(Prop.BACK_SUB_ROI).isDefault:
+            return workspace
+
+        self.log().information("Performing background subtraction")
+        args = {
+            "InputWorkspaceIndexType": "WorkspaceIndex",
+            "ProcessingInstructions": self.getPropertyValue(Prop.BACK_SUB_ROI),
+            "BackgroundCalculationMethod": "PerDetectorAverage",
+        }
+
+        try:
+            return self._run_algorithm(workspace, self._BACK_SUB_ALG, "InputWorkspace", args)
+        except Exception as ex:
+            # The error that's printed can be confusing if we don't mention the background subtraction algorithm
+            self.log().error(f"Error running {self._BACK_SUB_ALG}")
+            raise RuntimeError(ex)
+
+    def _create_transmission_ws(self, workspace):
+        """Create the transmission workspace"""
+        self.log().information("Creating the transmission workspace")
+        args = {
+            Prop.TRANS_ROI: self.getPropertyValue(Prop.TRANS_ROI),
+            Prop.I0_MON_IDX: self.getPropertyValue(Prop.I0_MON_IDX),
+            Prop.MON_WAV_MIN: self.getPropertyValue(Prop.MON_WAV_MIN),
+            Prop.MON_WAV_MAX: self.getPropertyValue(Prop.MON_WAV_MAX),
+        }
+        return self._run_algorithm(workspace, self._TRANS_WS_ALG, "FirstTransmissionRun", args)
+
+    def _run_algorithm(self, workspace, alg_name: str, input_ws_prop_name: str, args: dict):
+        """Run the specified algorithm as a child algorithm using the arguments provided.
+        The input and output workspace properties are both set to be the provided workspace."""
+
+        def run_alg(ws):
+            args.update({input_ws_prop_name: ws, "OutputWorkspace": ws})
+            alg = self.createChildAlgorithm(alg_name, **args)
+            alg.setRethrows(True)
```

Comment:
I think if the algorithm is a child algorithm, it will rethrow anyway. So this line may not be necessary.

Reply:
I had always understood it to be the opposite actually, that a child algorithm won't rethrow automatically. I might test to double check at some point, but I seem to recall finding that from previous experience, so I'll keep this in if that's OK.

Reply:
I looked it up on the algorithm.cpp class: 
```      
if (m_isChildAlgorithm || m_runningAsync || m_rethrow) 
       throw;
```
Maybe there are other parts of the algorithms that interact with it too, or I don't know if it works differently in data processor algorithms. In any case, as it is is not hurting anyone.

Reply:
I found an old comment from Martyn on Slack that said child algorithms would set this automatically. I may have got confused by some behaviour I saw in some of our unit tests quite a while ago. It looks like the Reflectometry code does use `setRethrows` with child algorithms quite often (I noticed it quite a bit in `ReflectometryISISLoadAndProcess`), but I think you're right that it's probably not necessary. As you say, doesn't really do any harm, but one to remember going forward, so thanks for flagging it up!